packages/frontend/src/components/SimpleTable/DashboardCellContextMenu.tsx

```diff
@@ -59,34 +58,36 @@ const DashboardCellContextMenu: FC<
     const filterField =
         isDimension(item) && !item.hidden
             ? [
-                  {
-                      id: uuid4(),
-                      target: {
-                          fieldId: fieldId(item),
-                          tableName: item.table,
-                      },
-                      operator: FilterOperator.EQUALS,
-                      values: [value.raw],
-                      label: undefined,
-                  },
+                  createDashboardFilterRuleFromField({
+                      field: item,
+                      availableTileFilters: {},
+                      isTemporary: true,
+                      value: value.raw,
+                  }),
               ]
             : [];
 
     const possiblePivotFilters = (
         meta?.pivotReference?.pivotValues || []
-    ).map<DashboardFilterRule>((pivot) => {
+    ).reduce<DashboardFilterRule[]>((acc, pivot) => {
         const pivotField = itemsMap?.[pivot?.field];
-        return {
-            id: uuid4(),
-            target: {
-                fieldId: pivot.field,
-                tableName: isField(pivotField) ? pivotField?.table : '',
-            },
-            operator: FilterOperator.EQUALS,
-            values: [pivot.value],
-            label: undefined,
-        };
-    });
+        if (
+            !pivotField ||
+            !isField(pivotField) ||
+            !isFilterableField(pivotField)
+        )
```

Comment:
can you explain what is the reason behind this change?

Reply:
oh because the `pivotField` has to be filterable (of this type Field | Dimension | Metric | TableCalculationField) 

When creating a Dashboard Filter Rule it has to be `FilterableField`, otherwise we don't allow it. This will work as before, just more type robust

Reply:
thanks!